### PR TITLE
Refactor Export Transaction

### PR DESCRIPTION
**Pull request by amanalili**
#### PR Requirements

If you have made **any** visual changes to the GUI. Make sure you have:
- [ ] Tested on iOS Tablet
- [ ] Tested on small Android

### CODE REVIEW

File: src/components/scenes/TransactionsExportScene.js [hunk 1]
```diff
@@ -24,22 +32,86 @@ type StateProps = {
 
 type Props = StateProps & PassedProps
 
-export class TransactionsExportSceneComponent extends Component<Props> {
```
Comment (amanalili) [inline src/components/scenes/TransactionsExportScene.js:27]: Use PureComponent

File: src/components/scenes/TransactionsExportScene.js [hunk 2]
```diff
@@ -24,22 +32,86 @@ type StateProps = {
...
+  endDate: Date,
+  isExportQbo: boolean,
+  isExportCsv: boolean
+}
+
+export class TransactionsExportSceneComponent extends Component<Props, State> {
+  constructor(props: Props) {
+    super(props)
+    this.state = {
+      startDate: new Date(Date.now() - 1000 * 60 * 60 * 24),
```
Comment (amanalili) [inline src/components/scenes/TransactionsExportScene.js:65]: use Last Month as a default

File: src/components/scenes/TransactionsExportScene.js [hunk 3]
```diff
@@ -24,22 +32,86 @@ type StateProps = {
...
+
+  setLastMonth = () => {
+    const lastMonth = new Date(new Date().setMonth(new Date().getMonth() - 1))
+    this.setState({
+      startDate: new Date(new Date().getFullYear(), lastMonth.getMonth(), 1, 0, 0, 0),
+      endDate: new Date(new Date().getFullYear(), new Date().getMonth(), 1, 0, 0, 0)
+    })
+  }
+
+  export = async () => {
```
Comment (amanalili) [inline src/components/scenes/TransactionsExportScene.js:92]: exportFile

File: src/components/scenes/TransactionsExportScene.js [hunk 4]
```diff
@@ -24,22 +32,86 @@ type StateProps = {
...
+      await this.exportQBO()
+    }
+    if (this.state.isExportCsv) {
+      await this.exportCSV()
+    }
+  }
+
   render() {
+    const { startDate, endDate, isExportCsv, isExportQbo } = this.state
+    const walletName = `${this.props.sourceWallet.name || 'My Wallet'} (${this.props.currencyCode})`
```
Comment (amanalili) [inline src/components/scenes/TransactionsExportScene.js:121]: "No Wallet Name" in italics

File: src/components/scenes/TransactionsExportScene.js [hunk 5]
```diff
@@ -84,34 +79,66 @@ export class TransactionsExportSceneComponent extends Component<Props, State> {
...
+                  <PrimaryButton onPress={this.export} disabled={disabledExport}>
+                    <PrimaryButton.Text>{s.strings.string_export}</PrimaryButton.Text>
+                  </PrimaryButton>
+                </View>
+              )}
             </View>
-          )}
+          </TouchableWithoutFeedback>
         </ScrollView>
+        {datePicker !== null && Platform.OS === 'ios' && (
```
Comment (amanalili) [inline src/components/scenes/TransactionsExportScene.js:149]: remove Platform.OS

File: src/locales/en_US.js [hunk 6]
```diff
@@ -697,7 +697,9 @@ const strings = {
   string_end: 'End',
   export_transaction_quickbooks_qbo: 'Quickbooks QBO',
   export_transaction_csv: 'CSV',
-  string_export: 'Export'
+  string_export: 'Export',
+  export_transaction_error_1: 'Starting date should be earlier than the ending date',
```
Comment (amanalili) [inline src/locales/en_US.js:701]: Start date should be earlier than the end date

File: src/components/scenes/TransactionsExportScene.js [hunk 7]
```diff
@@ -73,6 +74,13 @@ export class TransactionsExportSceneComponent extends Component<Props, State> {
   }
 
   export = async () => {
+    const { startDate, endDate } = this.state
+    if (startDate.getTime() > endDate.getTime()) {
+      return showError(s.strings.export_transaction_error_1)
+    }
+    if (endDate.getTime() > Date.now()) {
```
Comment (amanalili) [inline src/components/scenes/TransactionsExportScene.js:97]: Remove error on end date > now

File: src/locales/en_US.js [hunk 8]
```diff
@@ -699,7 +699,8 @@ const strings = {
   export_transaction_csv: 'CSV',
   string_export: 'Export',
   export_transaction_error_1: 'Starting date should be earlier than the ending date',
-  export_transaction_error_2: 'Ending date should not be later than now'
+  export_transaction_error_2: 'Ending date should not be later than now',
+  export_transaction_default_wallet_name: 'MyWallet'
```
Comment (amanalili) [inline src/locales/en_US.js:703]: No Wallet Name

File: src/components/scenes/TransactionsExportScene.js (hunk 1 above)
Comment (amanalili) [reply src/components/scenes/TransactionsExportScene.js:27]: updated

File: src/components/scenes/TransactionsExportScene.js (hunk 2 above)
Comment (amanalili) [reply src/components/scenes/TransactionsExportScene.js]: updated

File: src/components/scenes/TransactionsExportScene.js (hunk 3 above)
Comment (amanalili) [reply src/components/scenes/TransactionsExportScene.js]: updated

File: src/components/scenes/TransactionsExportScene.js (hunk 4 above)
Comment (amanalili) [reply src/components/scenes/TransactionsExportScene.js]: updated

File: src/components/scenes/TransactionsExportScene.js (hunk 5 above)
Comment (amanalili) [reply src/components/scenes/TransactionsExportScene.js]: updated

File: src/locales/en_US.js (hunk 6 above)
Comment (amanalili) [reply src/locales/en_US.js]: updated

File: src/components/scenes/TransactionsExportScene.js (hunk 7 above)
Comment (amanalili) [reply src/components/scenes/TransactionsExportScene.js]: updated

File: src/locales/en_US.js (hunk 8 above)
Comment (amanalili) [reply src/locales/en_US.js]: updated

File: src/components/scenes/TransactionsExportScene.js (hunk 2 above)
Comment (amanalili) [reply src/components/scenes/TransactionsExportScene.js]: Put last month const on constructor

File: src/locales/en_US.js [hunk 9]
```diff
@@ -705,6 +705,8 @@ const strings = {
   string_no_wallet_name: 'No Wallet Name',
   export_transaction_error: 'Start date should be earlier than the end date',
   export_transaction_export_error: 'No transactions in the date range chosen',
+  export_transaction_export_error_2: 'Select only one export type on android',
```
Comment (paullinator) [inline src/locales/en_US.js:708]: This string should not be needed. On Android force the user to not set more than one export type.

File: src/components/scenes/TransactionsExportScene.js [hunk 10]
```diff
@@ -91,10 +110,14 @@ export class TransactionsExportSceneComponent extends PureComponent<Props, State
...
   exportFile = async () => {
-    const { startDate, endDate } = this.state
+    const { startDate, endDate, isExportQbo, isExportCsv } = this.state
     if (startDate.getTime() > endDate.getTime()) {
       return showError(s.strings.export_transaction_error)
     }
+    if (Platform.OS === 'android' && isExportQbo && isExportCsv) {
+      showError(s.strings.export_transaction_export_error_2)
+      return
+    }
```
Comment (paullinator) [inline src/components/scenes/TransactionsExportScene.js:120]: This should not be needed. The UI should prevent the user from selecting both export types.

**Pull request merged**